src/Compat/FormDecorator/PrimaryButtonDecorator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace ipl\Web\Compat\FormDecorator;
+
+use ipl\Html\Attribute;
+use ipl\Html\Contract\DecorationResult;
+use ipl\Html\Contract\Form;
+use ipl\Html\Contract\FormDecoration;
+use ipl\Html\Contract\FormSubmitElement;
+
+class PrimaryButtonDecorator implements FormDecoration
+{
+    public function decorateForm(DecorationResult $result, Form $form): void
+    {
+        $primaryButtonCandidate = null;
+        foreach ($form->getElements() as $element) {
+            if (! $element instanceof FormSubmitElement) {
+                continue;
+            }
+
+            /** @var string[] $classes */
+            $classes = (array) $element->getAttributes()->get('class')->getValue();
+
+            // If a class attribute's value is defined as a string such as 'a b c' or as an array like ['a b c'],
+            // it is not checked for existing 'btn-*' or 'btn-primary' classes.
+            if (count($classes) === 1 && str_contains($classes[0], ' ')) {
+                $primaryButtonCandidate = $element;
+                break;
+            }
```

Comment:
Ehm, not sure this "optimization" is necessary.

Reply:
Without this condition, a value starting with `btn-a test` will skip the final if check because of the expression `! array_find($classes, fn($v) => str_starts_with($v, 'btn-'))`, and the class is not added.